core/src/test/java/org/elasticsearch/search/SearchHitsTests.java

```diff
@@ -42,7 +42,7 @@ public static SearchHits createTestItem() {
         for (int i = 0; i < searchHits; i++) {
             hits[i] = SearchHitTests.createTestItem(false); // creating random innerHits could create loops
         }
-        long totalHits = randomLong();
+        long totalHits = frequently() ? Math.abs(randomLong()) : -1;
```

Comment:
Math.abs(randomLong()) can be negative if the random number generator returns MIN_VALUE, maybe use TestUtil.nextLong?